hosted/eks/p0/p0_provisioning_test.go

```diff
@@ -21,148 +21,65 @@ import (
-				success, err := clusters.CheckServiceAccountTokenSecret(ctx.RancherClient, clusterName)
-				Expect(err).To(BeNil())
-				Expect(success).To(BeTrue())
-			})
-
-			By("checking all management nodes are ready", func() {
-				err := nodestat.AllManagementNodeReady(ctx.RancherClient, cluster.ID, helpers.Timeout)
-				Expect(err).To(BeNil())
-			})
-
-			By("checking all pods are ready", func() {
-				podErrors := pods.StatusPods(ctx.RancherClient, cluster.ID)
-				Expect(podErrors).To(BeEmpty())
-			})
-
-			currentNodeGroupNumber := len(cluster.EKSConfig.NodeGroups)
-			initialNodeCount := *cluster.EKSConfig.NodeGroups[0].DesiredSize
-
-			By("scaling up the NodeGroup", func() {
-				var err error
-				cluster, err = helper.ScaleNodeGroup(cluster, ctx.RancherClient, initialNodeCount+1)
-				Expect(err).To(BeNil())
-				err = clusters.WaitClusterToBeUpgraded(ctx.RancherClient, cluster.ID)
-				Expect(err).To(BeNil())
-				for i := range cluster.EKSConfig.NodeGroups {
-					Expect(*cluster.EKSConfig.NodeGroups[i].DesiredSize).To(BeNumerically("==", initialNodeCount+1))
-				}
-			})
-
-			By("scaling down the NodeGroup", func() {
+			BeforeEach(func() {
 				var err error
-				cluster, err = helper.ScaleNodeGroup(cluster, ctx.RancherClient, initialNodeCount)
-				Expect(err).To(BeNil())
-				err = clusters.WaitClusterToBeUpgraded(ctx.RancherClient, cluster.ID)
-				Expect(err).To(BeNil())
-				for i := range cluster.EKSConfig.NodeGroups {
-					Expect(*cluster.EKSConfig.NodeGroups[i].DesiredSize).To(BeNumerically("==", initialNodeCount))
-				}
-			})
+				GinkgoLogr.Info("Using K8s version: " + k8sVersion)
+				eksConfig := new(eks.ClusterConfig)
+				config.LoadAndUpdateConfig(eks.EKSClusterConfigConfigurationFileKey, eksConfig, func() {
+					eksConfig.Region = region
+					eksConfig.Tags = helper.GetTags()
+					eksConfig.KubernetesVersion = &k8sVersion
+				})
 
-			By("adding a NodeGroup", func() {
-				var err error
-				cluster, err = helper.AddNodeGroup(cluster, increaseBy, ctx.RancherClient)
+				cluster, err = eks.CreateEKSHostedCluster(ctx.RancherAdminClient, clusterName, ctx.CloudCred.ID, false, false, false, false, map[string]string{})
 				Expect(err).To(BeNil())
-				err = clusters.WaitClusterToBeUpgraded(ctx.RancherClient, cluster.ID)
+				cluster, err = helpers.WaitUntilClusterIsReady(cluster, ctx.RancherAdminClient)
 				Expect(err).To(BeNil())
-				Expect(len(cluster.EKSConfig.NodeGroups)).To(BeNumerically("==", currentNodeGroupNumber+1))
 			})
-			By("deleting the NodeGroup", func() {
-				var err error
-				cluster, err = helper.DeleteNodeGroup(cluster, ctx.RancherClient)
-				Expect(err).To(BeNil())
-				err = clusters.WaitClusterToBeUpgraded(ctx.RancherClient, cluster.ID)
-				Expect(err).To(BeNil())
-				Expect(len(cluster.EKSConfig.NodeGroups)).To(BeNumerically("==", currentNodeGroupNumber))
-
+			AfterEach(func() {
+				// TODO Check if EKS cluster deleted on AWS
```

Comment:
**nit:**
This can be removed now that we have setup janitors.

Reply:
Removed, along other TODOs which are already implemented